### PR TITLE
WFLY-16532 [Preview] elytron-oidc-client: add ability to configure additional scope for authentication request

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
Analysis doc: https://github.com/wildfly/wildfly-proposals/pull/527
WFLY Issue: https://issues.redhat.com/browse/WFLY-16532
Related ELY Issue: https://issues.redhat.com/browse/ELY-2574
Depends on: https://github.com/wildfly-security/wildfly-elytron/pull/1925
Depends on: https://github.com/wildfly/wildfly/pull/17705

This PR supersedes https://github.com/wildfly/wildfly/pull/17785. It includes a fix for an [error](https://ci.wildfly.org/buildConfiguration/WF_PullRequest_WildFlyPreviewLinuxJdk17/427599?expandBuildDeploymentsSection=false&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true&expandBuildProblemsSection=false&showLog=427599_37316_460&logView=flowAware) that occurred in testsuite/integration/elytron due to incorrect stability levels.

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-19176](https://issues.redhat.com/browse/WFLY-19176)
> * [WFLY-19078](https://issues.redhat.com/browse/WFLY-19078)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

File: testsuite/integration/elytron/pom.xml [hunk 1]
```diff
@@ -618,6 +620,7 @@
                                         <jboss-maven-dist/>
                                         <jboss-fork-embedded>${galleon.fork.embedded}</jboss-fork-embedded>
                                         <optional-packages>passive+</optional-packages>
+                                        <config-stability-level>preview</config-stability-level>
```
Comment (fjuma) [inline testsuite/integration/elytron/pom.xml:623]: Is this needed?

Comment (PrarthonaPaul) [reply testsuite/integration/elytron/pom.xml]: removed!

File: testsuite/integration/elytron/pom.xml [hunk 2]
```diff
@@ -488,6 +490,7 @@
                         <jboss.inst>${wildfly.dir}</jboss.inst>
                         <module.path>${wildfly.dir}/modules</module.path>
                         <node0>${node0}</node0>
+                        <embedded.server.stability>preview</embedded.server.stability>
```
Comment (fjuma) [inline testsuite/integration/elytron/pom.xml:493]: Here, I think we can use `<embedded.stability>${embedded.server.stability}</embedded.stability>`.

File: testsuite/integration/elytron/src/test/resources/arquillian.xml [hunk 3]
```diff
@@ -13,7 +13,7 @@
         <configuration>
             <property name="javaVmArguments">${server.jvm.args}</property>
             <property name="serverConfig">${jboss.server.config.file.name:standalone.xml}</property>
-            <property name="jbossArguments">${jboss.args}</property>
+            <property name="jbossArguments">${jboss.args} --stability=${embedded.server.stability}</property>
```
Comment (fjuma) [inline testsuite/integration/elytron/src/test/resources/arquillian.xml:16]: And then this would be `${embedded.stability}`

Comment (PrarthonaPaul) [reply testsuite/integration/elytron/src/test/resources/arquillian.xml]: fixed!

**Review by fjuma — APPROVED** [review]
Adding my approval on this one. Will keep an eye on CI.

**Comment by bstansberry** [comment]
https://github.com/wildfly/wildfly/pull/17792 should fix the MPScriptTestCase failures

**Pull request merged**